src/tasks/withdraw/traded_tokens.ts

```diff
@@ -66,16 +80,34 @@ export async function getAllTradedTokens(
       throw new Error("Too many events in the same block");
     }
     const mid = Math.floor(((await numericToBlock) + fromBlock) / 2);
-    const { tokens: firstHalf } = await getAllTradedTokens(
-      settlement,
-      fromBlock,
-      mid,
-      hre,
-    );
-    const { tokens: secondHalf, toBlock: numberSecondHalf } =
-      await getAllTradedTokens(settlement, mid + 1, toBlock, hre);
-    tokens = [...firstHalf, ...secondHalf];
-    numericToBlock = numberSecondHalf;
+
+    const firstHalf = await getAllTradedTokens(settlement, fromBlock, mid, hre);
```

Comment:
Why don't we want to recover `partialTradedTokens` in the first half as well?

Reply:
We do already!
The idea is that if the error is in the first half, then we (continue to) throw the error that was created in `getAllTradedTokens`, which might contain a partial token list and doesn't need any change as no new information on the tokens was added at this point of the execution.
The important thing to notice is that the only way to get useful token information is having the first half succeed at some point in some recursion step. Then this info is attached to the error when executing the second half. If the first half always fails, then there cannot be any useful token information to be sent along with the error.